tests/instruments/hp/test_hp8116a.py

```diff
@@ -0,0 +1,266 @@
+        ret = HP8116A._parse_value_with_unit(case, self.UNITS)
+        assert math.isclose(ret, expected)
+
+    def test_generate_1_2_5_sequence(self):
+        min = 200e-6
+        max = 1
+        expected = [200e-6, 500e-6, 1e-3, 2e-3, 5e-3, 10e-3, 20e-3, 50e-3,
+                    100e-3, 200e-3, 500e-3, 1]
+        ret = HP8116A._generate_1_2_5_sequence(min, max)
+        assert ret == expected
+
+    @pytest.mark.parametrize('case', BOOLEANS)
+    def test_haversine_enabled(self, case, make_resetted_instr):
+        instr = make_resetted_instr
+        instr.haversine_enabled = case
+        assert instr.haversine_enabled == case
+
+    @pytest.mark.parametrize('case', BOOLEANS)
+    def test_autovernier_enabled(self, case, make_resetted_instr):
+        instr = make_resetted_instr
+        instr.autovernier_enabled = case
+        assert instr.autovernier_enabled == case
+
+    @pytest.mark.parametrize('case', BOOLEANS)
+    def test_limit_enabled(self, case, make_resetted_instr):
+        instr = make_resetted_instr
+        instr.limit_enabled = case
+        assert instr.limit_enabled == case
+
+    @pytest.mark.parametrize('case', BOOLEANS)
+    def test_complement_enabled(self, case, make_resetted_instr):
+        instr = make_resetted_instr
+        instr.complement_enabled = case
+        assert instr.complement_enabled == case
+
+    @pytest.mark.parametrize('case', BOOLEANS)
+    def test_output_enabled(self, case, make_resetted_instr):
+        instr = make_resetted_instr
+        instr.output_enabled = case
+        assert instr.output_enabled == case
+
+    @pytest.mark.parametrize('case, expected', FREQUENCIES)
+    def test_frequency(self, case, expected, make_resetted_instr):
+        instr = make_resetted_instr
+        instr.frequency = case
+        assert math.isclose(instr.frequency, expected)
+
+    def test_given_invalid_frequency_when_set_frequency_then_(self, make_resetted_instr):
+        instr = make_resetted_instr
+        with pytest.raises(ValueError):
+            instr.frequency = 1e9  # Sadly 1 GHz is too high :(
+
+    def test_check_has_option_001(self, make_resetted_instr):
+        instr = make_resetted_instr
+        assert instr._check_has_option_001() == self.HAS_OPTION_001
+
+    def test_given_resetted_when_check_errors_then_no_error(self, make_resetted_instr):
+        instr = make_resetted_instr
+        errors = instr.check_errors()
+        assert type(errors) is list
+        assert len(errors) == 0
+
+    def test_given_invalid_duty_cycle_when_check_errors_then_error(self, make_resetted_instr):
+        instr = make_resetted_instr
+        instr.frequency = 50e6
+        instr.duty_cycle = 90
+        errors = instr.check_errors()
+        assert len(errors) == 1
+        assert errors[0] == 'DUTY C. ERROR'
+
+    def test_given_two_error_conditions_when_check_errors_then_two_errors(self,
+                                                                          make_resetted_instr):
```

Comment:
Self descriptive function name :)

Reply:
Yeah "we" as computer scientists / software engineers (not electrical engineers just doing some programming) sometimes do that :joy: (Not meant to be offensive)